server/schemas/schemaModel.js

```diff
@@ -13,31 +13,31 @@ bluebird.promisifyAll(Model);
 
 module.exports = {
   model: Model,
+  /*
+  createTemplate:
+    Accepts incoming request, response, and next (callback).
+    Attempts to find an existing schema with the provided name. If one does not exists,
+    a schema template is created from the request parameters.
 
+  */
   createTemplate: function(request, response, next) {
     var error;
-    var newSchema = request.body.schema;
-    if (!newSchema.name) {
-      error = {message: "Invalid schema name!"};
-      next(error, false);
-      return;
-    }
-    Model.findOneAsync({name: newSchema.name})
+    var newSchema = request.body;
+    var params = {
```

Comment:
it seems like `params` is actually the new schema, and `newSchema` is just `body` or `requestBody` or something similar. but i might not understand what's going on...